diploma-project/pages/catalog/[id].jsx

```diff
@@ -1,34 +1,52 @@
 import {StandardLayout} from "../../components";
-import {default_profile_picture} from "../../public/constants";
+import {default_profile_picture, errorToast, successToast} from "../../public/constants";
 import Image from "next/future/image";
 import {useEffect, useState} from "react";
-import {getCatalog, Product} from "../../services/product";
+import {getCatalog} from "../../services/product";
 import {ProductCard} from "../../components";
 import {useRouter} from "next/router";
-import {getUserById} from "../../services/user";
+import {getUserById, UpdateFollowers} from "../../services/user";
+import {useUserStore} from "../../store/userStorage";
 
-const Catalog = ({user}) => {
+const Catalog = ({craftsman}) => {
+    const { currentUser } = useUserStore((state) => ({currentUser: state.user}));
+    const [isLogged, setIsLogged] = useState(false);
     const [products, setProducts] = useState(null);
+    const [toggleDrawerContent, setToggleDrawerContent] = useState(true);
+    const [craftsmanFollowers, setCraftsmanFollowers] = useState(craftsman.followers);
     const router = useRouter();
 
+    const onFollow = () => {
+        let followers = [...craftsmanFollowers];
+        const isFollower = craftsmanFollowers.includes(currentUser.uid);
+        if(isFollower) {
+            followers = followers.filter((uid) => uid !== currentUser.uid);
+        } else {
+            followers.push(currentUser.uid);
+        }
+
+        UpdateFollowers(craftsman.uid, followers)
+            .then(() => {
+                setCraftsmanFollowers(followers);
+                successToast(`Successfully ${isFollower ? "unfollowed" : "followed"} ${craftsman.name}!`);
+            })
+            .catch(() => {
+                errorToast(`Error: Couldn't ${isFollower ? "unfollow" : "follow"} ${craftsman.name}! Refresh the page and try again!`);
+        });
+    }
+
     useEffect(() => {
-        getCatalog(user.email).then((products) => {
-            setProducts(
-                products.map((product) => (
-                    new Product(
-                        product.title,
-                        product.description,
-                        product.displayImageURL,
-                        product.owner,
-                        product.price,
-                        product.createdDate,
-                        product.likes,
-                        product.tags
-                    )
-                ))
-            );
+        getCatalog(craftsman).then((catalog) => {
+            setProducts(catalog);
         });
-    }, [user.catalog]);
+    }, []);
```

Comment:
<img width="721" alt="Screenshot 2024-02-12 at 19 40 20" src="https://github.com/ItsRizee/diploma_project/assets/42142769/3f7b961a-ed22-4591-9c88-61f966945546">

You should really consider using a linter 😆